astropy/units/core.py

```diff
@@ -2587,34 +2587,35 @@ def def_unit(
 
 
 def _condition_arg(value):
-    """
-    Validate value is acceptable for conversion purposes.
+    """Validate value is acceptable for conversion purposes.
 
-    Will convert into an array if not a scalar, and can be converted
-    into an array
+    Will convert into an array if not a scalar or array-like, where scalars
+    and arrays can be python and numpy types, anything that defines
+    ``__array_namespace__`` or anything that has a ``.dtype`` attribute.
 
     Parameters
     ----------
-    value : int or float value, or sequence of such values
+    value : scalar or array-like
 
     Returns
     -------
-    Scalar value or numpy array
+    Scalar value or array
 
     Raises
     ------
     ValueError
         If value is not as expected
+
     """
-    if isinstance(value, (np.ndarray, float, int, complex, np.void)):
+    if (
+        isinstance(value, (np.ndarray, float, int, complex))
+        or hasattr(value, "dtype")
```

Comment:
Is it necessary to fast-path ndarray before this check? Does it actually provide a speed up?

Reply:
`ndarray` is first because all `Quantity.to(...)` goes through here. That said, it adds only 1% or so to a quantity conversion, but is a bit more visible if one gets a converter first and then applies to an array.

Anyway, my sense would be to keep it, but with a preconstructed type as you suggested above.

Reply:
SGTM. 🚢